src/PySpectrometer2.py

```diff
@@ -208,330 +252,289 @@ def snapshot(savedata):
-	elif use_picamera and keyPress == ord("g"):#Gain down
-			picamGain -= 1
-			if picamGain <=0:
-				picamGain = 0.0
-			picam2.set_controls({"AnalogueGain": picamGain})
-			print("Camera Gain: "+str(picamGain))								
+			cv2.putText(spectrum_vertical,"Framerate: "+str(cfps),(message_loc1,51),font,0.4,(0,255,255),1, cv2.LINE_AA)
+			cv2.putText(spectrum_vertical,saveMsg,(message_loc1,69),font,0.4,(0,255,255),1, cv2.LINE_AA)
+
+		#Second column
+		cv2.putText(spectrum_vertical,holdmsg,(message_loc2,15),font,0.4,(0,255,255),1, cv2.LINE_AA)
+		cv2.putText(spectrum_vertical,"Savgol Filter: "+str(savpoly),(message_loc2,33),font,0.4,(0,255,255),1, cv2.LINE_AA)
+		cv2.putText(spectrum_vertical,"Label Peak Width: "+str(mindist),(message_loc2,51),font,0.4,(0,255,255),1, cv2.LINE_AA)
+		cv2.putText(spectrum_vertical,"Label Threshold: "+str(thresh),(message_loc2,69),font,0.4,(0,255,255),1, cv2.LINE_AA)
+		cv2.imshow(title1,spectrum_vertical)
+
+		if dispWaterfall:
+			#stack the images and display the waterfall	
+			if waterfall_vertical is None:
+				waterfall_vertical = np.concatenate((messages, cropped, waterfall), axis=0)
+			else:
+				np.concatenate((messages, cropped, waterfall), out=waterfall_vertical, axis=0)
+
+			#dividing lines...
+			cv2.line(waterfall_vertical,(0,80),(frameWidth,80),(255,255,255),1)
+			cv2.line(waterfall_vertical,(0,160),(frameWidth,160),(255,255,255),1)
+			#Draw this stuff over the top of the image!
+			#Display a graticule calibrated with cal data
+			textoffset = 12
+
+			#vertical lines every whole 50nm
+			for positiondata in fifties:
+				for i in range(162,480):
+					if i%20 == 0:
+						cv2.line(waterfall_vertical,(positiondata[0],i),(positiondata[0],i+1),(0,0,0),2)
+						cv2.line(waterfall_vertical,(positiondata[0],i),(positiondata[0],i+1),(255,255,255),1)
+				cv2.putText(waterfall_vertical,str(positiondata[1])+'nm',(positiondata[0]-textoffset,475),font,0.4,(0,0,0),2, cv2.LINE_AA)
+				cv2.putText(waterfall_vertical,str(positiondata[1])+'nm',(positiondata[0]-textoffset,475),font,0.4,(255,255,255),1, cv2.LINE_AA)
+
+			cv2.putText(waterfall_vertical,calmsg1,(message_loc1,15),font,0.4,(0,255,255),1, cv2.LINE_AA)
+			
+			if use_picamera:
+				cv2.putText(waterfall_vertical,calmsg3,(message_loc1,33),font,0.4,(0,255,255),1, cv2.LINE_AA)
+				cv2.putText(waterfall_vertical,saveMsg,(message_loc1,51),font,0.4,(0,255,255),1, cv2.LINE_AA)
+				cv2.putText(waterfall_vertical,"Gain: "+str(picamGain),(message_loc1,69),font,0.4,(0,255,255),1, cv2.LINE_AA)
+			else:
+				cv2.putText(waterfall_vertical,calmsg2,(message_loc1,33),font,0.4,(0,255,255),1, cv2.LINE_AA)
+				cv2.putText(waterfall_vertical,calmsg3,(message_loc1,51),font,0.4,(0,255,255),1, cv2.LINE_AA)
+				cv2.putText(waterfall_vertical,saveMsg,(message_loc1,69),font,0.4,(0,255,255),1, cv2.LINE_AA)
+			
+			cv2.putText(waterfall_vertical,holdmsg,(message_loc2,15),font,0.4,(0,255,255),1, cv2.LINE_AA)
+
+			cv2.imshow(title2,waterfall_vertical)
+
+		while True:
```

Comment:
Put this in a loop so that if I hold down `t` or `g` to adjust the gain, it responds smoothly.